### DISCUSSION TITLE
AR Flutter coaching overlay

### DISCUSSION THREAD

**Discussion by douglasbianchi** [CariusLars/ar_flutter_plugin#19] [Q&A] [answered]
Excited to see the cross platform support for AR! Most of my AR dev experience was with ARKit on native iOS with Swift for only one specific use case which included plane detection and hit tests, thats about it. iOS has a coaching overlay for ARKit that demonstrates to the user at the start of an ar scene the steps for localizing and detecting a plane. Is this feature active in this plugin? Is there a corresponding feature for arcore? This is critical for user experience since most users are oblivious to the requirements for the system to establish a scene ('slower' motion, area coverage of plane, etc). Should an issue be created to add this?

Great work getting a cross platform plugin out to the public!

**Comment by CariusLars** [comment, marked as answer]
Hi @douglasbianchi , thanks for your input! The coaching overlay is currently disabled in the plugin, but I agree that it would be valuable to have an option to en- and disable it. For ARCore/Sceneform there also exists an instruction setting, even though it's not as comprehensive as the one in ARKit (see https://developers.google.com/sceneform/reference/com/google/ar/sceneform/ux/PlaneDiscoveryController). 
A good place for putting the option would be in the onInitialize method of the [sessionManager](https://github.com/CariusLars/ar_flutter_plugin/blob/main/lib/managers/ar_session_manager.dart) in my opinion. I think it should be quite easy to add the feature, it would be great if you opened a pull request that does so, I'm more than happy to review the code! Otherwise, feel free to open an issue as well :)
Cheers
Lars

**Reply by douglasbianchi** [reply]
Cool good to know! I am currently building out an app I am migrating from iOS. It might be a few weeks but ill take a stab at it. I'll probably circle back with some questions. Cheers! 😀

**Reply by CariusLars** [reply]
awesome!
